### PR TITLE
[export] Upstream support of (tensor, tensor list) in op returns.

### PR DESCRIPTION
**Pull request by zhxchen17**
Summary:
Upstreaming from internal to oss.
Diff: D49710320

Test Plan: buck2 build mode/opt sigmoid/inference/test_gpu:package_gen

Differential Revision: D50577490

cc @avikchaudhuri @gmagogsfm @tugsbayasgalan @angelayi

### CODE REVIEW

**Comment by SherlockNoMad** [comment]
test case plz :)

File: torch/_export/serde/serialize.py [hunk 1]
```diff
@@ -1364,6 +1428,34 @@ def deserialize_outputs(self, serialized_node: Node, fx_node: torch.fx.Node):
...
+                        "call_function",
+                        operator.getitem,
+                        (fx_node, idx),
+                        name=name,
+                    )
+                    self.sync_fx_node(name, individual_output)
+                    meta_val.append(self.serialized_name_to_meta[name])
+                    # The derived `getitem` nodes should have the same stacktrace as the
+                    # original `fx_node`
+                    individual_output.meta.update(self.deserialize_metadata(serialized_node.metadata))
```
Comment (SherlockNoMad) [inline torch/_export/serde/serialize.py:1445]: nit... 

`self.deserialize_metadata(serialized_node.metadata)` is returning the same result in line 1445 and 1454? 
This result can be cached in advance.

**Review by SherlockNoMad — APPROVED** [review]
per comments.

File: test/export/test_serialize.py [hunk 2]
```diff
@@ -236,7 +236,7 @@ def _check_graph_nodes(gm1, gm2, _check_meta=True):
                     elif isinstance(val1, (list, tuple)) and isinstance(val2, (list, tuple)):
                         # Or both are fake tensors lists with one element and with the
                         # same shape/dtype
-                        for v1, v2 in zip(val1, val2):
+                        for v1, v2 in zip(pytree.tree_flatten(val1)[0], pytree.tree_flatten(val2)[0]):
```
[the remark this answers is not in the document]
Comment (zhxchen17) [reply test/export/test_serialize.py:239]: pytree.tree_flatten() returns (flattend_data, tree_spec)...

**Pull request closed without merging**